purchases/src/main/kotlin/com/revenuecat/purchases/amazon/handler/PurchaseHandler.kt

```diff
@@ -95,70 +116,60 @@ internal class PurchaseHandler(
+                        // WIP: replace link with our own
+                        // Indicates that the purchase failed. Can simply mean user cancelled.
+                        //
+                        // According to Amazon's EU MFA flow documentation, the flow would
+                        // return a PurchaseResponse.RequestStatus.FAILED if cancelled and a
+                        // PurchaseResponse.RequestStatus.SUCCESSFUL if the MFA flow is completed successfully
+                        //
+                        // Amazon docs:
+                        // https://developer.amazon.com/blogs/appstore/post/
+                        // 72ee4001-9e54-49b1-83ad-f1cede9b40ce/
+                        // ensuring-your-implementation-of-in-app-purchase-is-ready-for-eu-multi-factor-authentication
+                        PurchasesError(PurchasesErrorCode.PurchaseCancelledError, AmazonStrings.ERROR_PURCHASE_FAILED)
+                    }
+                    PurchaseResponse.RequestStatus.INVALID_SKU -> PurchasesError(
+                        PurchasesErrorCode.ProductNotAvailableForPurchaseError,
+                        AmazonStrings.ERROR_PURCHASE_INVALID_SKU,
+                    )
+                    PurchaseResponse.RequestStatus.ALREADY_PURCHASED -> PurchasesError(
+                        PurchasesErrorCode.ProductAlreadyPurchasedError,
+                        AmazonStrings.ERROR_PURCHASE_ALREADY_OWNED,
+                    )
+                    PurchaseResponse.RequestStatus.NOT_SUPPORTED -> PurchasesError(
+                        PurchasesErrorCode.StoreProblemError,
+                        AmazonStrings.ERROR_PURCHASE_NOT_SUPPORTED,
+                    )
+                    else -> PurchasesError(
+                        PurchasesErrorCode.StoreProblemError,
+                        AmazonStrings.ERROR_PURCHASE_UNKNOWN,
+                    )
+                }
+
+                diagnosticsTrackerIfEnabled?.trackAmazonPurchaseAttempt(
+                    productId = storeProduct.id,
+                    requestStatus = response.requestStatus.name,
+                    errorCode = error?.code?.code,
+                    errorMessage = error?.message,
+                    responseTime = Duration.between(startTime, dateProvider.now),
+                )
 
-            callbacks?.let { (onSuccess, onError) ->
-                when (response.requestStatus) {
-                    PurchaseResponse.RequestStatus.SUCCESSFUL ->
-                        onSuccessfulPurchase(response.receipt, response.userData, onSuccess)
-                    PurchaseResponse.RequestStatus.FAILED -> onFailed(onError)
-                    PurchaseResponse.RequestStatus.INVALID_SKU -> onInvalidSku(onError)
-                    PurchaseResponse.RequestStatus.ALREADY_PURCHASED -> onAlreadyPurchased(onError)
-                    PurchaseResponse.RequestStatus.NOT_SUPPORTED -> onNotSupported(onError)
-                    else -> onUnknownError(onError)
```

Comment:
I refactored all this so it was simpler to track errors in one place. I also think these separate methods didn't make much sense IMO. Lmk what you think!